test/extended/baremetal/hosts.go

```diff
@@ -33,6 +34,29 @@ func baremetalClient(dc dynamic.Interface) dynamic.ResourceInterface {
 	return baremetalClient.Namespace("openshift-machine-api")
 }
 
+type FieldGetterFunc func(obj map[string]interface{}, fields ...string) (interface{}, bool, error)
+
+func expectField(host unstructured.Unstructured, nestedField string, fieldGetter FieldGetterFunc) o.Assertion {
+	fields := strings.Split(nestedField, ".")
+
+	value, found, err := fieldGetter(host.Object, fields...)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(found).To(o.BeTrue(), "baremetalhost field not found")
+	return o.Expect(value)
+}
+
+func expectStringField(host unstructured.Unstructured, nestedField string) o.Assertion {
```

Comment:
These wrappers simplify the expects and seems useful. But I wonder why I can not find any usage in the repo. Does nobody check nested fields in resources? or use different method?

Reply:
There's another allowed approach for types that cannot be imported, and it's to copy and paste the struct and decode it. In this case I found easier to adopt the other recommended approach, ie to use `unstructured` pkg